examples/chat/client/cpp/chat.cc

```diff
@@ -158,26 +156,25 @@ int main(int argc, char** argv) {
   std::string ca_cert = oak::ApplicationClient::LoadRootCert(absl::GetFlag(FLAGS_ca_cert));
   LOG(INFO) << "Connecting to Oak Application: " << address;
 
-  RoomToken room_access_token;
-  if (!absl::Base64Unescape(absl::GetFlag(FLAGS_room_access_token), &room_access_token)) {
-    LOG(FATAL) << "Failed to parse --room_access_token as base64";
+  std::string room_secret = absl::GetFlag(FLAGS_room_secret);
+  std::string private_key;
+
+  // If no room secret, for retreiving a private key, was provided, create a fresh private/public
+  // key pair, and store the private key in a file for later use. A secure key sharing mechanims is
+  // needed to share this file with other clients.
+  if (room_secret.empty()) {
+    private_key = oak::ApplicationClient::GenerateKeyPair();
+    oak::ApplicationClient::StorePrivateKey(private_key, "chat-room.key");
+  } else {
+    private_key = oak::ApplicationClient::LoadPrivateKey(room_secret);
   }
 
-  std::unique_ptr<Chat::Stub> stub;
-
-  // If no room access token was provided, create a fresh one, and print it out so that other
-  // clients may join the same room.
-  if (room_access_token.empty()) {
-    // TODO(#1357): Generate a public / secret key pair and use it to authenticate the client and
-    // label requests.
-    oak::NonceGenerator<64> generator;
-    auto room_access_token_bytes = generator.NextNonce();
-    room_access_token = std::string(room_access_token_bytes.begin(), room_access_token_bytes.end());
-    LOG(INFO) << "Join this room with --address=" << address
-              << " --room_access_token=" << absl::Base64Escape(room_access_token);
-  }
+  // Use the room's secret to sign the challenge required for authenticating the client.
+  oak::identity::SignedChallenge signed_challenge =
+      oak::ApplicationClient::Sign(private_key, "oak-challenge");
```

Comment:
Could you link to where the magic string is coming from? And perhaps make it a const?